airflow/api_fastapi/serializers/dag_run.py

```diff
@@ -0,0 +1,43 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from datetime import datetime
+
+from pydantic import BaseModel, Field
+
+from airflow.utils.state import DagRunState
+from airflow.utils.types import DagRunTriggeredByType, DagRunType
+
+
+class DAGRunResponse(BaseModel):
+    """DAG Run serializer for responses."""
+
+    dag_run_id: str | None = Field(alias="run_id")
+    dag_id: str
+    logical_date: datetime | None
+    start_date: datetime | None
+    end_date: datetime | None
+    data_interval_start: datetime | None
+    data_interval_end: datetime | None
+    last_scheduling_decision: datetime | None
+    run_type: DagRunType
+    state: DagRunState
+    external_trigger: bool
+    triggered_by: DagRunTriggeredByType
+    conf: dict
```

Comment:
I believe so. A user shouldn't need to send a second request just for the note text.

Reply:
+1. The legacy endpoint returns the note, so the new endpoint should also return it.

It's a special 'easy' case because I believe there's already a relationshiop and an association_proxy on the `content` from `dag_run_note` table. All that to say that we do not even need to manually join anything, `.note` should be accessible.

Maybe we can just pre-load the relationship to make the serialization step faster. In the select query through the `.options` method.

Reply:
Thanks. I am updating the code

Reply:
I just started looking into this and realised, dag_maker cannot add a note to dag_run. Even in the earlier endpoint, during POST call to create Dag run has a separate condition which checks presence of `note` and adds it using `association_proxy`. 

So, I should be first adding dag_run_note creation functionality to `dag_maker` and then proceed write a test. I should also keep in mind to add dag_run_note when migrating POST create_dag_run endpoint.

Reply:
For the `POST` I need we need to be iso-functional with the legacy endpoint. If it supports `POST` with a `note` then the new endpoint should too. If not then it's fine to not having it in the first iteration :)

Reply:
For now, I updated tests to just check if `note is None`. Once I add endpoint to trigger dag run, I'll update tests to include testing note with something other than None. 

I also added a linked PR: https://github.com/apache/airflow/pull/42800 where I added  DagRunNotePydantic class.

Reply:
You can just add manually a note to any of the dag model, to be able to test not `None` notes in this PR.

You can take a look at to see how you can create a Note on a dagrun:
```
@security.requires_access_dag("PUT", DagAccessEntity.RUN)
@action_logging
@provide_session
def set_dag_run_note(*, dag_id: str, dag_run_id: str, session: Session = NEW_SESSION) -> APIResponse:
    """Set the note for a dag run."""
    dag_run: DagRun | None = session.scalar(
        select(DagRun).where(DagRun.dag_id == dag_id, DagRun.run_id == dag_run_id)
    )
    if dag_run is None:
        error_message = f"Dag Run id {dag_run_id} not found in dag {dag_id}"
        raise NotFound(error_message)
    try:
        post_body = set_dagrun_note_form_schema.load(get_json_request_dict())
        new_note = post_body["note"]
    except ValidationError as err:
        raise BadRequest(detail=str(err))

    current_user_id = get_auth_manager().get_user_id()
    if dag_run.dag_run_note is None:
        dag_run.note = (new_note, current_user_id)
    else:
        dag_run.dag_run_note.content = new_note
        dag_run.dag_run_note.user_id = current_user_id
    session.commit()
    return dagrun_schema.dump(dag_run)

```

Reply:
All I had to do was add a note as type `str`, and sqlalchemy did its magic!  

Thanks @pierrejeambrun . I updated the PR with the test for dag_note(non-None)